overcooked_ai_py/agents/agent.py

```diff
@@ -507,33 +422,24 @@ def ml_action(self, state):
         counter_objects = self.mlp.mdp.get_counter_objects_dict(state, list(self.mlp.mdp.terrain_pos_dict['X']))
         pot_states_dict = self.mlp.mdp.get_pot_states(state)
 
-        # NOTE: this most likely will fail in some tomato scenarios
-        curr_order = state.curr_order
 
         if not player.has_object():
-
-            if curr_order == 'any':
-                ready_soups = pot_states_dict['onion']['ready'] + pot_states_dict['tomato']['ready']
-                cooking_soups = pot_states_dict['onion']['cooking'] + pot_states_dict['tomato']['cooking']
-            else:
-                ready_soups = pot_states_dict[curr_order]['ready']
-                cooking_soups = pot_states_dict[curr_order]['cooking']
+            ready_soups = pot_states_dict['ready']
+            cooking_soups = pot_states_dict['cooking']
 
             soup_nearly_ready = len(ready_soups) > 0 or len(cooking_soups) > 0
             other_has_dish = other_player.has_object() and other_player.get_object().name == 'dish'
 
             if soup_nearly_ready and not other_has_dish:
                 motion_goals = am.pickup_dish_actions(counter_objects)
             else:
-                next_order = None
-                if state.num_orders_remaining > 1:
-                    next_order = state.next_order
+                next_order = list(state.all_orders)[0]
```

Comment:
I think for all classes that only work for your version of the MDP it would be nice to have some assertion checks that make sure they're not being used on some of the unsupported MDPs. In particular in this instance, you could assert that len(state.all_orders) == 1 and that the only order is a 3-onion soup

Reply:
Sounds good. Added the check and the error message if attempting to do more than 3-onion-soup order